helpers/notification.js

```diff
@@ -0,0 +1,244 @@
+import db from '../models';
+import passportHelper from './passport';
+import mailSender from './utils/mail-sender';
+
+const {
+  User, Follows, Notification
+} = db;
+
+/**
+ * @author Samuel Niyitanga
+ * @exports NotificationHelper
+ * @class NotificationHelper
+ * @description Notification Helper
+ * */
+class NotificationHelper {
+  /**
+     * Set notification option
+     * @param {object} req - an object
+     * @return {object} Returns response
+     * @static
+     */
+  static async setOption(req) {
+    const { option } = req.params;
+    const user = await User.findOne({ where: { id: req.user.id } });
+    const notifsArray = user.dataValues.notification;
+    notifsArray.push(option);
+    const optionSet = await user.update({ notification: notifsArray });
+    return optionSet;
+  }
+
+  /**
+     * Unset notification option
+     * @param {object} req - an object
+     * @return {object} Returns response
+     * @static
+     */
+  static async UnsetOption(req) {
+    const { option } = req.params;
+    const user = await User.findOne({ where: { id: req.user.id } });
+    const notifsArray = user.dataValues.notification;
+    const optionIndex = notifsArray.indexOf(option);
+    notifsArray.splice(optionIndex, 1);
+    const optionUnsetted = await user.update({ notification: notifsArray });
+    return optionUnsetted;
+  }
+
+  /**
+     * Check if option exists
+     * @param {object} req - an object
+     * @param {object} res - an object
+     * @param {object} next - an object
+     * @return {boolean} Returns if true if it is valid else return false
+     * @static
+     */
+  static async isOptionAvailable(req, res, next) {
+    const user = await User.findOne({ where: { id: req.user.id } });
+    const index = user.dataValues.notification
+      .findIndex(notification => notification === req.params.option);
+    if (index !== -1) {
+      return res.status(422).send({ status: 422, Error: 'You already have this Option set' });
+    }
+    next();
+  }
+
+  /**
+     * Check if option exists
+     * @param {object} req - an object
+     * @param {object} res - an object
+     * @param {object} next - an object
+     * @return {boolean} Returns if true if it is valid else return false
+     * @static
+     */
+  static async isOptionIn(req, res, next) {
+    const user = await User.findOne({ where: { id: req.user.id } });
+    const index = user.dataValues.notification
+      .findIndex(notification => notification === req.params.option);
+    if (index === -1) {
+      return res.status(422).send({ status: 422, Error: 'You don\'t have this Option set' });
+    }
+    next();
+  }
+
+  /**
+     * Check if it is valid params
+     * @param {object} req - an object
+     * @param {object} res - an object
+     * @param {object} next - an object
+     * @return {boolean} Returns if true if it is valid else return false
+     * @static
+     */
+  static isValidOption(req, res, next) {
```

Comment:
For this function just create an array and check if the option is in that array, like `notificationArray`. Then you will get one `if`, to make our code clean

Reply:
Done